composeApp/src/commonMain/kotlin/org/ooni/probe/data/repositories/UrlRepository.kt

```diff
@@ -39,10 +39,13 @@ class UrlRepository(
     suspend fun createOrUpdateByUrl(models: List<UrlModel>): List<UrlModel> =
         withContext(backgroundContext) {
             database.transactionWithResult {
-                val existingModels =
-                    database.urlQueries.selectByUrls(models.filter { it.id == null }.map { it.url })
-                        .executeAsList()
-                        .mapNotNull { it.toModel() }
+                val urlsWithoutId = models.filter { it.id == null }.map { it.url }
+                val existingModels: List<UrlModel> =
+                    // Some lists are too large for a single SQL query
+                    urlsWithoutId.chunked(300) { urlsChunk ->
```

Comment:
where are we getting this limit from

Reply:
It's random. I only crashed with 3000 URLs, so I assumed 300 was ok for SQL. We know we're fine with the 222 from NMS. Do you think we should chunk by another value?

And it's not a limit, we're just batching the queries in chunks.

Reply:
The idea is to have a value that is proven to work.